libraries/AP_NavEKF2/AP_NavEKF2_OptFlowFusion.cpp

```diff
@@ -129,7 +129,9 @@ void NavEKF2_core::EstimateTerrainOffset()
 
             // calculate Kalman gain
             float SK_RNG = sq(q0) - sq(q1) - sq(q2) + sq(q3);
+#if 0            
```

Comment:
Range finder should only be used to either directly observe height for the main filter or to estimate the height of terrain wrt the EKF origin, not both. Any change to range finder use logic needs to be coordinated between NavEKF2_core::selectHeightForFusion() and NavEKF2_core::EstimateTerrainOffset() to ensure that this requirement is not violated.

Reply:
You can see here: https://github.com/ArduPilot/ardupilot/blob/8f7bf2137234ba55c88b4fc978202221fa7ed2a1/libraries/AP_NavEKF2/AP_NavEKF2_OptFlowFusion.cpp#L95 that current behaviour is to inhibit terrain height estimation if range finder is being used as the primary height source and optical flow is being used for navigationand not terrain height estimation.

Reply:
I don't understand why you have added a compiler directive to bypass calculation of this term.

Reply:
It was a test to demonstrate the problem. Not suggesting this is the real fix!

Reply:
As discussed, the inhibition logic is failing at the crucial point that the rangefinder data becomes active leading to a very large delta in the terrainState which causes the copter to think it is 15m in the air when it has hit the ground.